### DISCUSSION TITLE
Include forwarded headers by default when deploying .net web project to container apps

### DISCUSSION THREAD

**Discussion by mip1983** [Azure/azure-dev#3235] [Ideas]
I'm in the process of getting the deployment of an aspire application working and one thing I hit was that authentication wasn't working with OIDC, the redirect url it was sending was http rather than https. I've used 'azd infra synth' to generate bicep and yaml files for the project.

There's an environment variable for .NET apps in container environments: ASPNETCORE_FORWARDEDHEADERS_ENABLED which needs to be set to true for external web apps in container environments for things like OIDC to work ([link](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-8.0#forward-the-scheme-for-linux-and-non-iis-reverse-proxies)).

Maybe this would be a sensible thing to include by default? Unless this is something that would make more sense as a default in Aspire i.e. I think you can do this in there by putting:

```
var apiservice = builder.AddProject<Projects.ecoDriverAnalytics_Api>("ecodriver-analytics-api")
	.WithEnvironment("ASPNETCORE_FORWARDEDHEADERS_ENABLED", "true")
```

For example, but maybe it should do that under the skin? If so, I'll post this suggestion under the aspire repo.

**Comment by davidfowl** [comment]
I don't think this should be in AZD. We can automatically set this in aspire for .NET projects. Can you open an issue on http://github.com/dotnet/aspire

**Reply by mip1983** [reply]
Thank you, will do
